lib/detect-testing-library-utils.ts

```diff
@@ -81,153 +81,169 @@ export function detectTestingLibraryUtils<
       DEFAULT_FILENAME_PATTERN;
 
     // Helpers for Testing Library detection.
-    const helpers: DetectionHelpers = {
-      getTestingLibraryImportNode() {
-        return importedTestingLibraryNode;
-      },
-      getCustomModuleImportNode() {
-        return importedCustomModuleNode;
-      },
-      getTestingLibraryImportName() {
-        return getImportModuleName(importedTestingLibraryNode);
-      },
-      getCustomModuleImportName() {
-        return getImportModuleName(importedCustomModuleNode);
-      },
-      /**
-       * Determines whether Testing Library utils are imported or not for
-       * current file being analyzed.
-       *
-       * By default, it is ALWAYS considered as imported. This is what we call
-       * "aggressive reporting" so we don't miss TL utils reexported from
-       * custom modules.
-       *
-       * However, there is a setting to customize the module where TL utils can
-       * be imported from: "testing-library/module". If this setting is enabled,
-       * then this method will return `true` ONLY IF a testing-library package
-       * or custom module are imported.
-       */
-      getIsTestingLibraryImported() {
-        if (!customModule) {
-          return true;
-        }
+    const getTestingLibraryImportNode: DetectionHelpers['getTestingLibraryImportNode'] = () => {
```

Comment:
I'm defining `DetectionHelpers` outside so then I can both pass it to `EnhancedRuleCreate` and reuse it for functions definitions. Ideally I'd like to do the opposite: type each function definition, then extract the type of final `helper` object and pass that to `EnhancedRuleCreate` but I don't know if there is a way of doing that since the type should be extracted to an outer scope somehow. This works tho.